joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -374,12 +374,17 @@ void JointTrajectoryController::read_state_from_hardware(JointTrajectoryPoint &
   auto assign_point_from_interface =
     [&](std::vector<double> & trajectory_point_interface, const auto & joint_interface)
   {
+    trajectory_point_interface.resize(dof_);
```

Comment:
For some reason, it's not (see the PR description for where I think the size is being changed) which is why it's segfaulting

Reply:
@destogl managed to find the cause? We could open an investigation ticket but I'm keen on merging this soon

Reply:
Not sure if this info is useful or not, but I encountered this issue when doing the following (using the latest master):

1) load the JTC into inactivate state from the CLI using the --set-state configured option
2) unload the JTC immediately

The result was a segfault in every single case. No reactivation needed, as the controller was not activated at all. On the other hand, if we load it into unconfigured state, and then attempt to unload it, the segfault doesn't occur. 
Neither when we're loading the JTC into inactive state -> activating it -> deactivating it (JTC is back into inactive state) and only then unloading it.

Reply:
@VX792 Did this PR resolve the issue for you? If yes, do you mind creating a test that trigger the segfault? I no longer have access to the codebase that reproduce this issue :crying_cat_face:

Reply:
Nope, it didn't. It's entirely possible that we're talking about two different issues, since mine was 'solved' by a really quick fix.

The problem was that the traj_home_point_ptr_ variable is initially a nullptr, and it only gets initialized in the on_activate() function. If we're calling the on_cleanup() fn through a unload_controller request, the cleanup will attempt to call a traj_home_point_ptr_->update(), trying to access a member of a nullptr. And that's a huge segfault!

We can check if the pointer is null and simply return (it works!), but I know nothing about control theory so @destogl might come up with a better solution.